clifford/_layout.py

```diff
@@ -175,6 +179,177 @@ def construct_graded_mt(
     return sparse.COO(coords=coords, data=mult_table_vals, shape=(dims, dims, dims))
 
 
+def get_as_ga_vector_func(layout):
+    """
+    Returns a function that converts a scalar into a GA value vector
+    for the given algebra
+    """
+    scalar_index = layout._basis_blade_order.bitmap_to_index[0]
+    ndims = layout.gaDims
+    @_numba_utils.njit
+    def as_ga_value_vector(x):
+        op = np.zeros(ndims)
+        op[scalar_index] = x
+        return op
+    return as_ga_value_vector
+
+
+def get_overload_add(layout):
+    """
+    Returns an overloaded JITed function that works on
+    MultiVector value arrays
+    """
+    scalar_index = layout._basis_blade_order.bitmap_to_index[0]
+
+    def ga_add(x):
+        # dummy function to overload
+        pass
+
+    @overload(ga_add, inline='always')
+    def ol_ga_add(a, b):
+        if isinstance(a, (types.Integer, types.Float)) and isinstance(b, types.Array):
+            def impl(a, b):
+                op = b.astype(np.float32)
+                op[scalar_index] += a
+                return op
+            return impl
+        elif isinstance(a, types.Array) and isinstance(b, (types.Integer, types.Float)):
+            def impl(a, b):
+                op = a.astype(np.float32)
+                op[scalar_index] += b
+                return op
+            return impl
+        else:
+            def impl(a, b):
+                return a + b
+            return impl
+
+    return ga_add
+
+
+def get_overload_sub(layout):
+    """
+    Returns an overloaded JITed function that works on
+    MultiVector value arrays
+    """
+    scalar_index = layout._basis_blade_order.bitmap_to_index[0]
+
+    def ga_sub(x):
+        # dummy function to overload
+        pass
+
+    @overload(ga_sub, inline='always')
+    def ol_ga_sub(a, b):
+        if isinstance(a, (types.Integer, types.Float)) and isinstance(b, types.Array):
```

Comment:
```suggestion
        if isinstance(a, types.abstract.Number) and isinstance(b, types.Array):
```
throughout, probably.